src/mod/server.mod/servmsg.c

```diff
@@ -1464,15 +1476,10 @@ static int gotcap(char *from, char *msg) {
         putlog(LOG_SERV, "*", "SASL: put AUTHENTICATE %s",
             SASL_MECHANISMS[sasl_mechanism]);
         dprintf(DP_MODE, "AUTHENTICATE %s\n", SASL_MECHANISMS[sasl_mechanism]);
+        sasl_timeout_time = time(NULL) + sasl_timeout;
```

Comment:
This is not robust against clock adjustments. It would be better to either do:
```c
sasl_timeout_time = sasl_timeout;
/* in secondly() */
if (!--sasl_timeout) { /* timeout */ }
```
or use clock_gettime with CLOCK_MONOTONIC

Reply:
yes. fixed.